### DISCUSSION TITLE
Use Magento Translations for text checks

### DISCUSSION THREAD

**Discussion by norgeindian** [elgentos/magento2-cypress-testing-suite#115] [Ideas]
We're currently adapting the base setup for one of our projects.
As we don't have the Magento sample data there, it's obvious that we need to adapt selectors, identifiers and so on and that's totally fine.
But while doing this, we find very often, that we just need to replace the existing text with the translation.
So for example there is in `cypress/integration/luma/cart/cart.spec.js` the check for `You have no items in your shopping cart.`.
As the translation exists in the Magento system, would it be an idea to somehow get this translation and compare the given text to the translation instead of the fixed text?
I think, this would decrease the need for adapting the module for different projects.
What do you think?

**Comment by norgeindian** [comment]
@peterjaap , what do you think? Would that be an idea?
If you were fine with that in general, we would be happy to contribute and try to set that up.
This would then be a step more to plug & play, so that more people could use it without needing to adapt a lot.

**Reply by peterjaap** [reply]
Yes I'd be okay with something that automatically fetches the correct translation. What would you suggest to tackle this, technically?

**Reply by norgeindian** [reply]
@peterjaap, our approach would be to add a PHP script, which calls `\Magento\Framework\Translate::loadData` and writes the returned values into a file. As far as I see, this is the only possibility to get all existing translations of a project at one place.
Then we could use https://docs.cypress.io/api/commands/readfile to create a cypress function, which always reads the translated values from that file and provide that functionality to be usable in the tests.
What do you think of this approach?
